external-plugins/rehearse/plugin/handler/handler.go

```diff
@@ -170,26 +188,81 @@ func (h *GitHubEventsHandler) shouldActOnIssueComment(event *github.IssueComment
 	return false
 }
 
-func (h *GitHubEventsHandler) canUserRehearse(err error, org string, userName string, pr *github.PullRequest) bool {
+func (h *GitHubEventsHandler) canUserRehearse(org string, repo string, pr *github.PullRequest, userName string, changedFiles []string) (canUserRehearse bool, message string) {
+
+	owners, err := h.ownersClient.LoadRepoOwners(org, repo, pr.Base.Ref)
+	if err != nil {
+		log.WithError(err).Errorln(fmt.Sprintf("Could not load owners on org %s and repo %s", org, repo))
+		return false, ""
+	}
+
+	// if user is top level approver, then
+	// rehearsal is allowed
+	// even if the author is not an org member
+	// top level approvers is only a very limited circle of people
+	topLevelApprovers := owners.TopLevelApprovers()
+	for topLevelApprover := range topLevelApprovers {
+		if userName == topLevelApprover {
+			return true, ""
+		}
+	}
+
+	// if author is not a member of the org, then
+	// rehearsal is not allowed
+	// in order to avoid compromising the org with harmful PRs
 	isAuthorMember, err := h.ghClient.IsMember(org, pr.User.Login)
 	if err != nil {
 		log.WithError(err).Errorln("Could not validate PR author with the repo org")
-		return false
+		return false, ""
 	}
 	if !isAuthorMember {
 		log.Warnln("Pr author is not a member of the repo org")
-		return false
+		return false, ""
 	}
+
+	// if the user that issues rehearsal is not a member of the org, then
+	// rehearsal is not allowed
+	// in order to avoid denial of service type attacks
```

Comment:
nit: it could be for a number of different reasons - I think its enough to just say that rehearsal is not allowed.